dask/dataframe/core.py

```diff
@@ -1654,6 +1661,44 @@ def count(self, axis=None, split_every=False):
                 result.divisions = (self.columns.min(), self.columns.max())
             return result
 
+    @derived_from(pd.DataFrame)
+    def mode(self, axis=0, dropna=True, split_every=False):  # this is the series class
+        """Use a counter on each partition, sum the resulting counters, iterate over key,
+        value pairs until num_occurrences isn't equal to the first one"""
+        if axis == 1:
+            meta = self._meta_nonempty.count(axis=axis)
+            result = map_partitions(
+                M.mode,
+                self,
+                meta=meta,
+                token=self._token_prefix + "mode",
+                axis=axis,
+                dropna=True,
+                enforce_metadata=False,
+            )
+            return result
+        else:
+            from collections import Counter
+
+            def _mode_chunk(df, dropna=dropna):  # already defined elsewhere now
```

Comment:
Ideally would remove this to avoid the closure.

Reply:
removed